### PR TITLE
refactor: complete migration from npm to Bun across entire project

### PR DESCRIPTION
**Pull request by harshanandak**
- Updated package.json: prepare script now uses 'bun add -d'
- Updated CLAUDE.md: Changed package manager to Bun
- Updated README.md: All npm install → bun install, npx → bunx
- Updated CONTRIBUTING.md: npm install → bun install
- Updated QUICKSTART.md: All npm/npx → bun/bunx
- Updated docs/TOOLCHAIN.md: All npm/npx → bun/bunx throughout
- Updated bin/forge.js: All manual install instructions now use bun
  - execFileSync now uses 'bun add -d' instead of 'npm install -D'
  - All bunx install suggestions updated
  - Comments and usage instructions updated

Project now fully standardized on Bun for performance (3x faster than npm). All documentation and installation flows consistent.

<h2>Greptile Overview</h2>

<details><summary><h3>Greptile Summary</h3></summary>

This PR completes the migration from npm to Bun as the primary package manager across the entire project. All installation commands, documentation, and error messages have been updated to use `bun`/`bunx` instead of `npm`/`npx`.

**Major changes:**
- Updated all documentation files (CLAUDE.md, README.md, QUICKSTART.md, TOOLCHAIN.md, CONTRIBUTING.md) with Bun commands
- Migrated `bin/forge.js` installation flows and error messages to use `bun add` and `bunx`
- Updated `package.json` prepare script to use Bun
- Fixed test infrastructure issues from previous commits (git config pollution, Windows path validation, CI timing)
- Updated agent configuration files (.cursorrules, .clinerules, .windsurfrules, GEMINI.md)

**Issues found:**
- One inconsistency in `CLAUDE.md:94-95` where `npx forge setup` was not updated to `bunx forge setup` in the Toolchain section

**Test improvements:**
- Fixed global git config side effects in test fixtures (already addressed in commit 87c8c88)
- Added Windows path validation for cross-platform compatibility
- Increased timing tolerance for Windows CI environments
- Added `--force` flag to CI workflow for reliable fixture recreation

The migration is comprehensive and follows the project's stated goal of standardizing on Bun for 3x performance improvement over npm.
</details>

<details><summary><h3>Confidence Score: 4.5/5</h3></summary>

- This PR is safe to merge after fixing the minor inconsistency in CLAUDE.md
- The migration is thorough and well-executed across 17 files. Only one minor inconsistency found (npx vs bunx in CLAUDE.md). Test improvements are solid and address real cross-platform issues. All changes are low-risk documentation and command updates with no breaking logic changes.
- CLAUDE.md needs the Toolchain section updated from npx to bunx for consistency
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| CLAUDE.md | Updated build commands from npm to Bun, but missed updating `npx forge setup` references in Toolchain section |
| bin/forge.js | Comprehensively migrated all npm/npx commands to bun/bunx throughout installation flows and error messages |
| package.json | Updated prepare script to use `bun add -d lefthook` instead of npm |
| docs/TOOLCHAIN.md | Thoroughly updated all package manager references from npm/npx to bun/bunx, removed duplicate npx examples |
| test-env/automation/setup-fixtures.sh | Fixed global git config pollution by using local-only configuration and improved error handling |

</details>

</details>

<details><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant Dev as Developer
    participant Docs as Documentation Files
    participant Code as Source Code (bin/forge.js)
    participant Tests as Test Suite
    participant CI as GitHub Actions

    Note over Dev,CI: Bun Migration Process

    Dev->>Docs: Update CLAUDE.md build commands
    Dev->>Docs: Update README.md installation
    Dev->>Docs: Update QUICKSTART.md examples
    Dev->>Docs: Update TOOLCHAIN.md all references
    Dev->>Docs: Update agent configs (.cursorrules, etc.)
    
    Dev->>Code: Update bin/forge.js installation flows
    Code->>Code: Replace npm install → bun add
    Code->>Code: Replace npx → bunx
    Code->>Code: Update error messages
    Code->>Code: Update manual install instructions
    
    Dev->>Code: Update package.json prepare script
    
    Dev->>Tests: Fix test-env/automation/setup-fixtures.sh
    Tests->>Tests: Remove global git config pollution
    Tests->>Tests: Add local-only git configuration
    Tests->>Tests: Improve error handling
    
    Dev->>Tests: Update edge case tests
    Tests->>Tests: Add Windows path validation
    Tests->>Tests: Increase timing tolerance for CI
    Tests->>Tests: Add cross-platform security checks
    
    Dev->>CI: Update .github/workflows/test.yml
    CI->>Tests: Add --force flag for fixture setup
    
    CI->>Tests: Run test suite on Linux/macOS/Windows
    Tests-->>CI: All tests pass
    
    Note over Dev,CI: Migration Complete: npm → Bun
```
</details>

### CODE REVIEW

File: test-env/automation/setup-fixtures.sh [hunk 1]
```diff
@@ -93,16 +93,45 @@ init_git_repo() {
...
 
+  # Configure git globally if not set (CI environments)
+  if ! git config --global user.email > /dev/null 2>&1; then
+    git config --global user.email "test@example.com" > /dev/null 2>&1 || true
+  fi
+  if ! git config --global user.name > /dev/null 2>&1; then
+    git config --global user.name "Test User" > /dev/null 2>&1 || true
+  fi
+  # Set default branch name to avoid warnings
+  git config --global init.defaultBranch main > /dev/null 2>&1 || true
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply test-env/automation/setup-fixtures.sh]: **Fixed in commit 87c8c88**

Excellent catch! Removed all global git config writes to avoid CI environment pollution.

**Changes:**
- ❌ Removed: `git config --global user.email/user.name/init.defaultBranch`
- ✅ Changed: `git init -b main` (sets branch locally, not globally)
- ✅ Kept: Local git config only (`git config user.email/user.name`)

**Impact:**
- No global environment mutation
- No side effects across CI jobs, repos, or concurrent test runs
- Fixtures still initialize correctly with local config

All tests still passing on Linux, macOS, and Windows.

Commit: https://github.com/harshanandak/forge/commit/87c8c88
